BatteryParameterTest.cpp

```diff
@@ -0,0 +1,36 @@
+#define CATCH_CONFIG_MAIN
+#include "test/catch.hpp"
+#include"BatteryParameter.h"
+#include"RandNumGenerator.h"
+
+
+
+TEST_CASE(" Test For Battery Current Readings..."){
+  
+	//Arrange..
+	int minCurrent = 0;    //cureent...
+	int maxCurrent = 100;
+	
+	int minSoc = 0;        //Soc...
+	int maxSoc = 50;
+
+	int GenerateNumbers = 150;
+	BatteryParameter batteryparameter(new randomNumberGenerator);
+
+	//Act...
+	bool currentexpectedResult = batteryparameter.currentGenerator( maxCurrent, GenerateNumbers);
+
+	//Assert...
+	assert(currentexpectedResult == true);
+	REQUIRE(batteryparameter.RefreshSignalContainer() == true);
+	assert(batteryparameter.inRange(minCurrent, maxCurrent) == true);

```

Comment:
These `inRange` asserts will check a random number generated. They can pass even when the range is wrong.
Focus on testing your functionality - the most crucial is the output format.
In the absence of a test for the output format, any mistakes will only be caught only in integration.